sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java

```diff
@@ -435,7 +438,9 @@ public void onFailure(Throwable e) {
             });
 
         }
-        if (!concreteTableParameter.settings().getAsMap().isEmpty() && indices.length > 0) {
+        if (!concreteTableParameter.settings().getAsMap().isEmpty()
+                && indices.length > 0
+                && !analysis.excludePartitions()) {
```

Comment:
if the indices are empty (see above) the  excludePartition check here isn't required